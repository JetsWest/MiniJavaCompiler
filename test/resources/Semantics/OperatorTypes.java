class SomeClass {
    public static void main(String[] args) {
        System.out.println(new A());
    }
}

class Object { }

class A {
    int[] arr;
    boolean bool1;
    boolean bool2;
    int num1;
    int num2;
    Object obj;

    public int and() {
        boolean x;

        x = true && false;
        x = bool1 && bool2;
        x = true && bool1;
        x = bool1 && true;

        x = true && arr;
        x = true && 1;
        x = true && num1;
        x = true && obj;

        return 0;
    }

    public int lessThan() {
        boolean x;

        x = 1 < 2;
        x = num1 < 2;
        x = 1 < num2;
        x = num1 < num2;
        x = num1 < arr[0];

        x = 1 < arr;
        x = 1 < bool1;
        x = 1 < obj;

        return 0;
    }

    public int plus() {
        int x;

        x = 1 + 2;
        x = num1 + 2;
        x = 1 + num2;
        x = num1 + num2;
        x = arr[0] + arr[1];

        x = 1 + arr;
        x = 1 + bool1;
        x = 1 + obj;

        return 0;
    }

    public int minus() {
        int x;

        x = 1 - 2;
        x = num1 - 2;
        x = 1 - num2;
        x = num1 - num2;
        x = arr[0] - arr[1];

        x = 1 - arr;
        x = 1 - bool1;
        x = 1 - obj;

        return 0;
    }

    public int times() {
        int x;

        x = 1 * 2;
        x = num1 * 2;
        x = 1 * num2;
        x = num1 * num2;
        x = arr[0] * arr[1];

        x = 1 * arr;
        x = 1 * bool1;
        x = 1 * obj;

        return 0;
    }
}