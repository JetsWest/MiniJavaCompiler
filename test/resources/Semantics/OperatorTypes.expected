(Line 24) Cannot compare non-boolean variables.
(Line 25) Cannot compare non-boolean variables.
(Line 26) Cannot compare non-boolean variables.
(Line 27) Cannot compare non-boolean variables.
(Line 42) Cannot compare non-integer values.
(Line 43) Cannot compare non-integer values.
(Line 44) Cannot compare non-integer values.
(Line 58) Cannot add non-integer values.
(Line 59) Cannot add non-integer values.
(Line 60) Cannot add non-integer values.
(Line 74) Cannot subtract non-integer values.
(Line 75) Cannot subtract non-integer values.
(Line 76) Cannot subtract non-integer values.
(Line 90) Cannot multiply non-integer values.
(Line 91) Cannot multiply non-integer values.
(Line 92) Cannot multiply non-integer values.